domain/src/DomainListBuilder.php

```diff
@@ -29,46 +126,55 @@ public function getFormId() {
    */
   public function getOperations(EntityInterface $entity) {
     $operations = parent::getOperations($entity);
-    $destination = \Drupal::destination()->getAsArray();
+    $destination = $this->destinationHandler->getAsArray();
     $default = $entity->isDefault();
     $id = $entity->id();
 
-    // @TODO: permission checks.
-    if ($entity->status() && !$default) {
-      $operations['disable'] = array(
-        'title' => $this->t('Disable'),
-        'url' => Url::fromRoute('domain.inline_action', array('op' => 'disable', 'domain' => $id)),
-        'weight' => 50,
-      );
+    // If the user cannot edit domains, none of these actions are permitted.
+    $admin = $this->accessHandler->checkAccess($entity, 'update');
```

Comment:
isn't `$admin` just `$access` (or `$can_update`) in this case?
previously `$admin` was used in the context of assigned domains.
I guess more of a nitpick, than anything